src/DynamoPackages/PackageLoader.cs

```diff
@@ -438,6 +467,25 @@ public Package ScanPackageDirectory(string directory, bool checkCertificates)
             return null;
         }
 
+        /// <summary>
+        /// Check and get the version from the version string. Throw a library load exception if anything is wrong with the version
+        /// </summary>
+        /// <param name="version">the version string</param>
+        /// <param name="name">name of the package</param>
+        /// <param name="directory">package directory</param>
+        /// <returns>Returns a valid Version</returns>
+        private Version CheckAndGetPackageVersion(string version, string name, string directory)
+        {
+            try
+            {
+                return new Version(version);
```

Comment:
There is an `VersionUtilities.PartialParse()` ..not sure if that does anything else ...

Reply:
That method is still using the Version class. We might want to use it if we decide that we don't want to do a full compare of the package version.  Package version `1.0.0.1` is now newer than version `1.0.0.0`. I have noticed that the packages I have looked at usually only defines the version as `1.0.0`. That will automatically turn into `1.0.0.0` before comparison in my current implementation. Have we defined somewhere how we deal with packages versions? E.g. should the last number always be ignored?

Reply:
I'm confused as is often the case - AFAIK when you upload a package you can only specify major,minor,and revision - not build.

see:
![image](https://user-images.githubusercontent.com/508936/112213182-33118780-8bf4-11eb-9633-661be5f0e929.png)

I don't think we can really encounter a package that sets the last version number unless the header is modified manually.

Reply:
Thanks, I think we should ignore the last version number then and use the utility method we have.